src/code-view/internal.tsx

```diff
@@ -2,71 +2,84 @@
+import { createHighlight } from "./highlight";
 import { CodeViewProps } from "./interfaces";
 import styles from "./styles.css.js";
 
 const ACE_CLASSES = { light: "ace-cloud_editor", dark: "ace-cloud_editor_dark" };
 
-function getLineNumbers(content: string) {
-  return content.split("\n").map((_, n) => n + 1);
-}
-
 type InternalCodeViewProps = CodeViewProps & InternalBaseComponentProps;
 
+const textHighlight = createHighlight(new TextHighlightRules());
+
 export function InternalCodeView({
   content,
   actions,
   lineNumbers,
+  lineWrapping,
   highlight,
   ariaLabel,
   ariaLabelledby,
   __internalRootRef = null,
   ...props
 }: InternalCodeViewProps) {
-  const code = highlight ? highlight(content) : <span>{content}</span>;
   const baseProps = getBaseProps(props);
   const preRef = useRef<HTMLPreElement>(null);
   const darkMode = useCurrentMode(preRef) === "dark";
 
   const regionProps = ariaLabel || ariaLabelledby ? { role: "region" } : {};
 
+  // Create tokenized elements of the content.
+  const code = highlight ? highlight(content) : textHighlight(content);
+
   return (
     <div
-      className={styles.root}
+      className={clsx(darkMode ? ACE_CLASSES.dark : ACE_CLASSES.light, styles.root)}
       {...regionProps}
       {...baseProps}
       aria-label={ariaLabel}
       aria-labelledby={ariaLabelledby}
       ref={__internalRootRef}
     >
-      <div className={clsx(lineNumbers && styles["root-with-numbers"], actions && styles["root-with-actions"])}>
-        <Box color="text-status-inactive" fontSize="body-m">
-          {lineNumbers && (
-            <div className={styles["line-numbers"]} aria-hidden={true}>
-              {getLineNumbers(content).map((number) => (
-                <span key={number}>{number}</span>
-              ))}
-            </div>
-          )}
-        </Box>
-        <pre
-          ref={preRef}
-          className={clsx(
-            darkMode ? ACE_CLASSES.dark : ACE_CLASSES.light,
-            styles.code,
-            lineNumbers && styles["code-with-line-numbers"],
-            actions && styles["code-with-actions"]
-          )}
-        >
-          <Box color="inherit" variant="code" fontSize="body-m">
-            {code}
-          </Box>
-        </pre>
-        {actions && <div className={styles.actions}>{actions}</div>}
-      </div>
+      <table className={clsx(styles["code-table"])}>
+        <colgroup>
+          <col style={{ width: 1 } /* shrink to fit content */} />
+          <col style={{ width: "auto" }} />
+        </colgroup>
+        <tbody>
+          {Children.map(code.props.children, (child, index) => {
+            return (
+              <tr key={index}>
+                {lineNumbers && (
+                  <td className={styles["line-number"]} aria-hidden={true}>
+                    <Box color="text-status-inactive" fontSize="body-m">
+                      {index + 1}
+                    </Box>
```

Comment:
Changing the markup here breaks selection feature. Line numbers get added into selection.

For the reference you can check how it works on Github

https://github.com/cloudscape-design/code-view/assets/812240/c0bb70c2-1c31-4260-b304-4392d8dec5f3

Reply:
ah, yeah that's a good catch, thanks. Added some CSS that should fix this.